gitea: new options and settings refactor (#206179)
NixOS/nixpkgs: nixos/doc/manual/release-notes/rl-2305.section.md

- The option `services.prometheus.exporters.pihole.interval` does not exist anymore and has been removed.

- `k3s` can now be configured with an EnvironmentFile for its systemd service, allowing secrets to be provided without ending up in the Nix Store.

> - `gitea` module options have been made more RFC042 conforming (i.e. more options are now in `settings`)

Reviewer: > To avoid merge conflicts, consider adding your item at an arbitrary place in the list instead.

Author: Thanks, will do next time.

> - `gitea` module options have been made more RFC042 conforming (i.e. more options are now in `settings`)

Reviewer 2: I wouldn't expect an average NixOS admin who's not involved with upstream development at all to know what RFC42 is about (despite the nice number :p)

Hence I'd prefer to provide a few more details, i.e. that a bunch of options don't exist anymore, but can be set with `services.gitea.settings` and that the keys match the upstream config params now.

- The option `services.prometheus.exporters.pihole.interval` does not exist anymore and has been removed.

- `k3s` can now be configured with an EnvironmentFile for its systemd service, allowing secrets to be provided without ending up in the Nix Store.

> - `gitea` module options have been made more RFC042 conforming (i.e. some options were removed in favor the official ones by using `settings`)

Reviewer: ```suggestion
- `gitea` module options have been changed to be RFC042 conforming (i.e. some options were moved to be located under `services.gitea.settings`)
```

otherwise people thing, that things would have been fully removed.
also please see the comment below the above headline.
